testing/lib/mock_registry.py

```diff
@@ -0,0 +1,87 @@
+# Copyright 2017 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from containerregistry.client.v2_2 import docker_image
+import unittest
+
+
+class MockRegistry():
+
+    def __init__(self):
+        self._registry = {}
+        self._manifests = {}
+        self._tags = {}
+
+    def getRegistry(self):
+        return self
+
+    def getFullRepoStr(self, full_repo):
+        if isinstance(full_repo, docker_image.FromRegistry):
+            full_repo = full_repo.getName()
+        return str(full_repo)
+
+    def getRepoStr(self, repository):
```

Comment:
Can you re-use any of the existing __str__ methods in docker_name.py here?

Reply:
I looked, but couldn't find anything :(